src/SentryHandler.php

```diff
@@ -89,27 +91,32 @@ static function ($highest, $record) {
      */
     protected function write(array $record): void
     {
-        $sentryEvent = [
-            'level'   => $sentryLevel = $this->getSeverityFromLevel($record['level']),
-            'message' => (new LineFormatter('%channel%.%level_name%: %message%'))->format($record),
-        ];
+        $sentryEvent = SentryEvent::createEvent();
+        $sentryEvent->setLevel($sentryLevel = $this->getSeverityFromLevel((int) $record['level']));
+        $sentryEvent->setMessage((new LineFormatter('%channel%.%level_name%: %message%'))->format($record));
 
-        if (isset($record['context']['exception']) && $record['context']['exception'] instanceof \Throwable) {
-            $sentryEvent['exception'] = $record['context']['exception'];
+        if (isset($record['context']['exception']) && $record['context']['exception'] instanceof Throwable) {
+            $sentryEvent->setExceptions([new ExceptionDataBag($record['context']['exception'])]);
         }
 
-        $this->hub->withScope(function (Scope $scope) use ($record, $sentryEvent, $sentryLevel): void {
-            $scope->setLevel($sentryLevel);
-            $scope->setExtra('monolog.formatted', $record['formatted'] ?? '');
-
-            foreach ($this->breadcrumbsBuffer as $breadcrumbRecord) {
-                $scope->addBreadcrumb(new Breadcrumb(
-                    $this->getBreadcrumbLevelFromLevel($breadcrumbRecord['level']),
-                    $this->getBreadcrumbTypeFromLevel($breadcrumbRecord['level']),
-                    $breadcrumbRecord['channel'] ?? 'N/A',
-                    $breadcrumbRecord['formatted'] ?? 'N/A'
-                ));
-            }
+        $this->hub->withScope(
+            function (Scope $scope) use ($record, $sentryEvent, $sentryLevel): void {
+                $scope->setLevel($sentryLevel);
+                $scope->setExtra('monolog.formatted', $record['formatted'] ?? '');
+
+                foreach ($this->breadcrumbsBuffer as $breadcrumbRecord) {
+                    $context = array_merge($breadcrumbRecord['context'], $breadcrumbRecord['extra']);
```

Comment:
Probably `exception` should be remove from context :thinking: But if it's used properly, `exception` will be only in highest-level record, so it won't be in breadcrumbs I think.

Reply:
Not sure we need the context at all actually ^^
Why did you add it?

Reply:
Because Sentry can display it nicely which can be useful for debugging:

![image](https://user-images.githubusercontent.com/600668/177054057-df1a63b7-a469-469e-830a-a458c2aeb7a9.png)

Reply:
I need to try it on a real project, I know mine can have lots of context.

Don't have time for that so let's try it this way.

Reply:
Let's keep exception, it does not anything, cannot harm, just ugly.